contracts/DependencyRegistryV0.sol

```diff
@@ -0,0 +1,751 @@
+     * @return additionalRepositoryCount Count of additional repository URLs for dependency type
+     * @return projectWebsite Project website URL for dependency type
+     * @return availableOnChain Whether dependency type is available on chain
+     * @return scriptCount Count of on-chain scripts for dependency type
+     */
+    function dependencyTypeDetails(bytes32 _dependencyTypeId)
+        external
+        view
+        returns (
+            string memory typeAndVersion,
+            string memory preferredCDN,
+            uint24 additionalCDNCount,
+            string memory preferredRepository,
+            uint24 additionalRepositoryCount,
+            string memory projectWebsite,
+            bool availableOnChain,
+            uint24 scriptCount
+        )
+    {
+        DependencyType storage dependencyType = dependencyTypeInfo[
+            _dependencyTypeId
+        ];
+
+        return (
+            _dependencyTypeId.toString(),
+            dependencyType.preferredCDN,
+            dependencyType.additionalCDNCount,
+            dependencyType.preferredRepository,
+            dependencyType.additionalRepositoryCount,
+            dependencyType.projectWebsite,
+            dependencyType.scriptCount > 0,
+            dependencyType.scriptCount
+        );
+    }
+
+    /**
+     * @notice Overrides the script type and version that 
+     * would be returned by the core contract (`_contractAddress`)
+     * for a given project  (`projectId`) with the given dependency
+     * type (`_dependencyTypeId`).
+     * @param _contractAddress Core contract address.
+     * @param _projectId Project to override script type and version for.
+     * @param _dependencyTypeId Dependency type to return for project.
+     */
+    function addProjectDependencyOverride(
+        address _contractAddress,
+        uint256 _projectId,
+        bytes32 _dependencyTypeId
+    ) external onlyAdminACL(this.addProjectDependencyOverride.selector) {
+        require(
+            _dependencyTypes.contains(_dependencyTypeId),
+            "Dependency type is not registered"
+        );
+        projectDependencyOverrides[_contractAddress][
+            _projectId
+        ] = _dependencyTypeId;
+
+        emit ProjectDependencyOverrideAdded(
+            _contractAddress,
+            _projectId,
+            _dependencyTypeId
+        );
+    }
+
+    /**
+     * @notice Removes the script type and version override for a given 
+     * project (`projectId`) on a given core contract (`_contractAddress`).
+     * @param _contractAddress Core contract address.
+     * @param _projectId Project to remove override for.
+     */
+    function removeProjectDependencyOverride(
+        address _contractAddress,
+        uint256 _projectId
+    ) external onlyAdminACL(this.addProjectDependencyOverride.selector) {
+        delete projectDependencyOverrides[_contractAddress][_projectId];
```

Comment:
input validation to ensure that the override exists would be good here

Reply:
++